Sources/FoundationEssentials/Predicate/KeyPath+Inspection.swift

```diff
@@ -53,8 +55,9 @@ extension AnyKeyPath {
         case 1: // struct/tuple/self stored property
             fallthrough
         case 3: // class stored property
-            // Stored property components are each one word
-            if header._keyPathHeader_bufferSize > Self.WORD_SIZE {
+            // Stored property components are either just the payload, or the payload plus 32 bits if the payload is the sentinel value
+            let size = (firstComponentHeader._keyPathComponentHeader_payload == .PROPERTY_OFFSET_TOO_LARGE) ? MemoryLayout<UInt64>.size : MemoryLayout<UInt32>.size
+            if header._keyPathHeader_bufferSize > size {
```

Comment:
This is a bug fix that our tests started hitting with this change. For stored properties, there is always a component header (32 bits) and if the offset index doesn't fit inside of the header, the header is followed by a second 32 bits that contain the offset. Additionally, components are all word-aligned. Previously, we just checked if the buffer size was greater than one word (because on 64 bit platforms a single component is either 32 bits of header + 32 bits of padding or 32 bits of header and 32 bits of offset). But on watchOS which is 64_32, a word is only 32 bits so a single component is either just 32 bits of header (no padding) or 32 bits of header + 32 bits of offset. The updated arithmetic here accounts for this by comparing to 64/32 bits depending on whether the extra 32 bits of offset is used rather than comparing to word size.